src/devices/GrovePi/Sensors/DigitalInput.cs

```diff
@@ -0,0 +1,71 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Iot.Device.GrovePiDevice.Models;
+using System;
+using System.Collections.Generic;
+using System.Device.Gpio;
+
+namespace Iot.Device.GrovePiDevice.Sensors
+{
+    /// <summary>
+    /// DigitalInput is a generic calss for digital input
+    /// </summary>
+    public class DigitalInput : ISensor<PinValue>
+    {
+        internal GrovePi _grovePi;
+
+        /// <summary>
+        /// 
+        /// </summary>
+        /// <param name="grovePi">The GrovePi class</param>
+        /// <param name="port">The grove Port, need to be in the list of SupportedPorts</param>
+        public DigitalInput(GrovePi grovePi, GrovePort port)
+        {
+            if (!SupportedPorts.Contains(port))
+                throw new ArgumentException($"Grove port {port} not supported.", nameof(port));
+            _grovePi = grovePi;
+            Port = port;
+            _grovePi.PinMode(Port, PinMode.Input);
+        }
+
+        /// <summary>
+        /// Get the pin level, either high either low
+        /// </summary>
+        public PinValue Value => _grovePi.DigitalRead(Port);
+
+        /// <summary>
+        /// Returns the value as a string
+        /// </summary>
+        /// <returns>Returns the value as a string</returns>
+        public override string ToString() => Value.ToString();
+
+        /// <summary>
+        /// Get the name Digital Input
+        /// </summary>
+        public string SensorName => "Digital Input";
+
+        /// <summary>
+        /// grove sensor port
+        /// </summary>
+        public GrovePort Port { get; internal set; }
```

Comment:
In general we haven't been exposing the pins, controllers, etc. passed to drivers. This should probably be an internal field.